extra/datasets/__init__.py

```diff
@@ -35,9 +35,15 @@ def _load_disk_tensor(sz, bs, db_list, path, shuffle=False):
     return X, Y
   fn = Path(__file__).parent.resolve() / "cifar-10-python.tar.gz"
   download_file('https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz', fn)
-  tt = tarfile.open(fn, mode='r:gz')
-  db = [pickle.load(tt.extractfile(f'cifar-10-batches-py/data_batch_{i}'), encoding="bytes") for i in range(1,6)]
-  X_train, Y_train = _load_disk_tensor(50000, 10000, db, "cifar_train", shuffle=shuffle)
-  db = [pickle.load(tt.extractfile('cifar-10-batches-py/test_batch'), encoding="bytes")]
-  X_test, Y_test = _load_disk_tensor(10000, 10000, db, "cifar_test", shuffle=shuffle)
+  tar = fn.with_suffix('')
+  if fn.suffix == ".gz" and not tar.exists():
+    with gzip.open(fn, 'rb') as read_handle:
```

Comment:
I really wonder whether there is a more elegant way to do this from python (tar.gz -> tar), but could not find one in `gzip`, `tarfile` or `shutil`.